### PR TITLE
Dynamic import MySqlHook/PostgresHook in CloudSQLDatabaseHook

### LINKED ISSUE #40175: Google provider : Unnecessary imports for CloudSQL operators
### Apache Airflow Provider(s)

google

### Versions of Apache Airflow Providers

google, ssh, slack

### Apache Airflow version

2.6.3

### Operating System

composer

### Deployment

Google Cloud Composer

### Deployment details

_No response_

### What happened

When trying to load the dag for a test with a CloudSQLExportInstanceOperator (from airflow.providers.google.cloud.operators.cloud_sql) I have a module not found error (ModuleNotFoundError: No module named 'airflow.providers.mysql').

I do not use mysql, so I should not have to install this module.

### What you think should happen instead

We should not need those dependencies if not necessary. Moreover it is only used for type checking in another operator (CloudSQLExecuteQueryOperator)

### How to reproduce

Import a CloudSQL operator without the mysql provider installed for example

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by jhongy1994**
closes: #40175 

CloudSQLDatabaseHook: Dynamic imports  MySqlHook/PostgresHook within get_database_hook method

Changes:
- Implement dynamic imports within the get_database_hook method to import hooks only when needed.
- Change the return type from `PostgresHook | MySqlHook` to `BaseHook`.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by moiseenkov — COMMENTED** [review]
Changes look good, but what's the reason for placing imports inside methods? Could we just keep them on the top?

**Comment by jhongy1994** [comment]
@moiseenkov Thank you for reviewing!

The issue #40175 : ModuleNotFoundError (`ModuleNotFoundError: No module named 'airflow.providers.mysql'`) occurs when calling the cloud_sql Operator (e.g. `CloudSQLExportInstanceOperator`) even when MySqlHook is not used.

By placing the module import inside the method, we ensure that the module is imported only when it is actually needed, thereby avoiding the import error when the module is not required.

The first screenshot shows what happens when the import is placed at the top, without the mysql module.
image

The second screenshot shows what happens when the import is placed inside the method.
image

**Review by moiseenkov — APPROVED** [review]

File: airflow/providers/google/cloud/hooks/cloud_sql.py [hunk 1]
```diff
@@ -1140,22 +1138,28 @@ def get_sqlproxy_runner(self) -> CloudSqlProxyRunner:
...
         Retrieve database hook.
 
         This is the actual Postgres or MySQL database hook that uses proxy or
         connects directly to the Google Cloud SQL database.
         """
         if self.database_type == "postgres":
-            db_hook: PostgresHook | MySqlHook = PostgresHook(connection=connection, database=self.database)
+            from airflow.providers.postgres.hooks.postgres import PostgresHook
+
+            db_hook: BaseHook = PostgresHook(connection=connection, database=self.database)
```
Comment (shahar1) [inline airflow/providers/google/cloud/hooks/cloud_sql.py:1151]: Do you get a mypy error without `: BaseHook`?

**Review by shahar1 — APPROVED** [review]
Welcome to Apache AIrflow, it looks good!
I got a small question before merging :)

File: airflow/providers/google/cloud/hooks/cloud_sql.py (hunk 1 above)
Comment (jhongy1994) [reply airflow/providers/google/cloud/hooks/cloud_sql.py:1151]: @shahar1 Thank you for your review!!

Right. I got a mypy error when `: BaseHook` is missing.
Below is a screenshot of the error I encountered when running pre-commit without `: BaseHook`.
(pre-commit hook id: `mypy-providers`)
image

**Pull request merged**